src/Microsoft.AspNetCore.Routing/Internal/UriBuilderContextPooledObjectPolicy.cs

```diff
@@ -11,7 +11,12 @@ public class UriBuilderContextPooledObjectPolicy : IPooledObjectPolicy<UriBuildi
     {
         private readonly UrlEncoder _encoder;
 
-        public UriBuilderContextPooledObjectPolicy(UrlEncoder encoder)
+        public UriBuilderContextPooledObjectPolicy()
+            : this(UrlEncoder.Default)
+        {
+        }
+
+        internal UriBuilderContextPooledObjectPolicy(UrlEncoder encoder)
```

Comment:
Unless the single test using this overload fails if it users the other (parameterless) one, do not add this constructor.